### PR TITLE
WIP: stats: bins=auto in docstrings

### PR DESCRIPTION
**Pull request by ev-br**
Try improving histograms in the docstrings of distributions.

E.g. in master, https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.lognorm.html there's a single bin; let's see if it's going to get better with bin autoselection.

### CODE REVIEW

**Comment by ev-br** [comment]
TBH I'm baffled by the doc build failure. It works ok locally and the change looks harmless, but somehow some distribution (not sure which) hangs CircleCI.

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -199,7 +199,8 @@
 
 And compare the histogram:
 
->>> ax.hist(r, density=True, histtype='stepfilled', alpha=0.2)
+>>> ax.hist(r, density=True, bins='auto', histtype='stepfilled', alpha=0.2)
+>>> ax.set_xlim([x[0], x[-1]])
```
Comment (mdhaber) [inline scipy/stats/_distn_infrastructure.py:203]:
```suggestion
>>> # ax.set_xlim([x[0], x[-1]])
```
Let's see which line does it.

**Comment by mdhaber** [comment]
Let's see if merging main helps, since CIrcleCI doesn't do that.

**Comment by mdhaber** [comment]
Well that is wild.

**Comment by ev-br** [comment]
> Do you plan to dig into this @ev-br?

Thanks for looking into it Matt. Sadly, I won't be able to look into this CircleCI weirdness, in the next few weeks at least.

**Comment by mdhaber** [comment]
There was an error in the doc builds that didn't report as an error here on GitHub. It looks different than before, so rerunning in the off-chance that it was a fluke.

**Comment by mdhaber** [comment]
@tacaswell Tom, this seemingly harmless change causes our doc build to fail. Do you have any insight into how this might happen?

**Comment by mdhaber** [comment]
I wrote some code to try to track this down. I'm pretty sure `levy` and `levy_l` are the culprits. I'll see if I can change their parameters to make them nicer.

File: scipy/stats/_continuous_distns.py [hunk 2]
```diff
@@ -5026,7 +5026,50 @@ class levy_gen(rv_continuous):
...
+    >>> np.allclose([0.001, 0.5, 0.999], levy.cdf(vals))
+    True
+
+    Generate random numbers:
+
+    >>> r = levy.rvs(size=1000)
+
+    And compare the histogram:
+
+    >>> r = r[r <= x[-1]]
```
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:5068]: Oops. This essentially truncates the distribution. Need to normalize the histogram manually, I guess?

Comment (ev-br) [reply scipy/stats/_continuous_distns.py:5068]: This looks strange to me actually. Why doing this at all?

Comment (mdhaber) [reply scipy/stats/_continuous_distns.py:5068]: The tail is super long and fat. The huge random variates are what gives `hist` auto-binning trouble. Moreover, if you don't zoom in, all you see is a spike. So somehow we want to zoom in on the relevant region and avoid passing all those points to `hist`. There could be other ways, but this is one way.

Comment (ev-br) [reply scipy/stats/_continuous_distns.py:5068]: Ah, makes sense! Maybe then give this one a separate doc string example, not the general one? 
(I sort of wonder if e.g. cauchy not have the same problem)

Comment (mdhaber) [reply scipy/stats/_continuous_distns.py:5068]: Not as bad, apparently.
I plan to fix the normalization and add comments if I have to do anything unusual (like this), then merge. I'm just trying to finish up dormant PRs. But you're welcome to make-better in a follow-up!

**Comment by tacaswell** [comment]
Sorry, I missed the ping.  This appears to be sorted?

File: scipy/stats/_continuous_distns.py (hunk 2 above)
Comment (ev-br) [reply scipy/stats/_continuous_distns.py:5068]: Sounds good to me!

**Comment by mdhaber** [comment]
Well I have a workaround, but we had trouble here because it looks like `hist` with `bins='auto'` is causing hanging/crashing despite there being only 1000 data points. (Memory hogging, maybe?) Is that to be expected, or should I file a bug report?

**Comment by tacaswell** [comment]
Even if there are only 1k points, how many bins are being generated by the auto algorithm?  My guess is that there are a very large number of empty bins and we are handling that badly.

Can you open an issue with the example?  Either it is something we will be able to fix or serve as a marker for why we can not switch to "auto" as the default binning.

**Comment by mdhaber** [comment]
Refguide failure looks unrelated.
(`scipy.sparse.linalg.svds`)

File: scipy/stats/_continuous_distns.py [hunk 3]
```diff
@@ -5026,7 +5026,53 @@ class levy_gen(rv_continuous):
...
+    >>> import matplotlib.pyplot as plt
+    >>> fig, ax = plt.subplots(1, 1)
+
+    Calculate the first four moments:
+
+    >>> mean, var, skew, kurt = levy.stats(moments='mvsk')
+
+    Display the probability density function (``pdf``):
+
+    >>> # `levy` is very-heavy tailed.
```
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:5042]:
```suggestion
    >>> # `levy` is very heavy-tailed.
```

File: scipy/stats/_continuous_distns.py [hunk 4]
```diff
@@ -5083,7 +5129,53 @@ class levy_l_gen(rv_continuous):
...
+    >>> import matplotlib.pyplot as plt
+    >>> fig, ax = plt.subplots(1, 1)
+
+    Calculate the first four moments:
+
+    >>> mean, var, skew, kurt = levy_l.stats(moments='mvsk')
+
+    Display the probability density function (``pdf``):
+
+    >>> # `levy_l` is very-heavy tailed.
```
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:5145]:
```suggestion
    >>> # `levy_l` is very heavy-tailed.
```

**Comment by mdhaber** [comment]
This is an improvement, and CI is finally passing, so let's merge it. Thanks @ev-br!

**Comment by lobpcg** [comment]
> Refguide failure looks unrelated. (`scipy.sparse.linalg.svds`)
> 
> ```
> ===================
> scipy.sparse.linalg
> ===================
> 
> Non-deprecated objects in __all__: 33
> Objects in refguide: 33
> 
> scipy.sparse.linalg.svds
> ------------------------
> 
> File "build/testenv/lib/python3.9/site-packages/scipy/sparse/linalg/_eigen/_svds.py", line 239, in svds
> Failed example:
>     (np.allclose(s3, s) and
>      np.allclose(np.abs(u3), np.abs(u.toarray())) and
>      np.allclose(np.abs(vT3), np.abs(vT.toarray())))
> Expected:
>     True
> Got:
>     False
> ```

That hit me too out of the blue while I am working on https://github.com/scipy/scipy/pull/16712 I do not know if it is a coincidence, but I am making changes there to fix this. @WarrenWeckesser

**Pull request merged**